src/uiprotect/data/base.py

```diff
@@ -505,24 +520,34 @@ def _inject_api(
         data["api"] = api
         data_set = set(data)
```

Comment:
Ideally we would refactor everything to now have to inject the api object everywhere, but that would be a major breaking change